apis/control/v1alpha1/cassandratask_types.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TODO Labels to add that indicate the status of the task? So they can be searched? Like jobs have.
+
+// CassandraTaskSpec defines the desired state of CassandraTask
+type CassandraTaskSpec struct {
+
+	// ScheduledTime indicates the earliest possible time this task is executed. This does not necessarily
+	// equal to the time it is actually executed (if other tasks are blocking for example)
+	// +optional
+	ScheduledTime *metav1.Time `json:"scheduledTime,omitempty"`
+
+	// TODO BackOffLimit for RetryPolicy and maxMissedSeconds deadline for ScheduledTime?
+
+	// Which datacenter this task is targetting. Note, this must be a datacenter which the current cass-operator
+	// can access
+	Datacenter corev1.ObjectReference `json:"datacenter,omitempty"`
+
+	// Jobs defines the jobs this task will execute (and their order)
+	Jobs []CassandraJob `json:"jobs,omitempty"`
```

Comment:
Curious, what is the use case for multiple jobs?

Reply:
One of those "what ifs". It doesn't really add any extra code to the processing itself (a single for clause). I guess you could do something like cleanup + repair?